src/modules/accounting/accounting.py

```diff
@@ -327,3 +312,103 @@ def get_extra_data(self, blockstamp: ReferenceBlockStamp) -> ExtraData:
+        rebase_part = self._calculate_rebase_report(blockstamp)
+        modules_part = self._get_newly_exited_validators_by_modules(blockstamp)
+        wq_part = self._calculate_wq_report(blockstamp)
+
+        # Distinct parts explicitly labeled by version
+        # So at the top level it is clear in which parts the reports will differ
+        extra_data_part_v1 = self._calculate_extra_data_report_v1(blockstamp)
+        return self._combine_report_parts(1, blockstamp, rebase_part, modules_part, wq_part, extra_data_part_v1)
+
+    def _calculate_report_v2(self, blockstamp: ReferenceBlockStamp):
+        rebase_part = self._calculate_rebase_report(blockstamp)
+        modules_part = self._get_newly_exited_validators_by_modules(blockstamp)
+        wq_part = self._calculate_wq_report(blockstamp)
+
+        extra_data_part_v2 = self._calculate_extra_data_report_v2(blockstamp)
+        return self._combine_report_parts(2, blockstamp, rebase_part, modules_part, wq_part, extra_data_part_v2)
+
+    # fetches validators_count, cl_balance, withdrawal_balance, el_vault_balance, shares_to_burn
+    def _calculate_rebase_report(self, blockstamp: ReferenceBlockStamp) -> tuple[int, Gwei, Wei, Wei, int]:
+        validators_count, cl_balance = self._get_consensus_lido_state(blockstamp)
+        withdrawal_vault_balance = self.w3.lido_contracts.get_withdrawal_balance(blockstamp)
+        el_rewards_vault_balance = self.w3.lido_contracts.get_el_vault_balance(blockstamp)
+        shares_requested_to_burn = self.get_shares_to_burn(blockstamp)
+        return validators_count, cl_balance, withdrawal_vault_balance, el_rewards_vault_balance, shares_requested_to_burn
+
+    # calculates is_bunker, finalization_share_rate, finalization_batches
+    def _calculate_wq_report(self, blockstamp: ReferenceBlockStamp) -> tuple[bool, int, list[int]]:
+        is_bunker = self._is_bunker(blockstamp)
+        finalization_share_rate, finalization_batches = self._get_finalization_data(blockstamp)
+        return is_bunker, finalization_share_rate, finalization_batches
+
+    def _calculate_extra_data_report_v1(self, blockstamp: ReferenceBlockStamp) -> ExtraData:
+        stuck_validators, exited_validators, orl = self._get_generic_extra_data(blockstamp)
+        return ExtraDataService.collect(
+            stuck_validators,
+            exited_validators,
+            orl.max_items_per_extra_data_transaction,
+            orl.max_node_operators_per_extra_data_item,
+        )
+
+    def _calculate_extra_data_report_v2(self, blockstamp: ReferenceBlockStamp) -> ExtraData:
+        stuck_validators, exited_validators, orl = self._get_generic_extra_data(blockstamp)
+        return ExtraDataServiceV2.collect(
+            stuck_validators,
+            exited_validators,
+            orl.max_items_per_extra_data_transaction,
+            orl.max_node_operators_per_extra_data_item,
+        )
+
+    @staticmethod
+    def _update_metrics(report_data: ReportData):
```

Comment:
Can you also check if there are some other Metrics that could be moved here?
Looks like a good place to collect metrics across the whole report.